modules/core/src/main/java/org/apache/ignite/configuration/IgniteConfiguration.java

```diff
@@ -562,11 +566,24 @@ public IgniteConfiguration(IgniteConfiguration cfg) {
      * @return Optional grid name. Can be {@code null}, which is default grid name, if
      *      non-default grid name was not provided.
      */
+    @Deprecated
     public String getGridName() {
         return gridName;
     }
 
     /**
+     * Gets optional local instance name. Returns {@code null} if non-default local instance
+     * name was not provided.
+     *
+     * @return Optional local instance name. Can be {@code null}, which is default local
+     * instance name, if non-default local instance name was not provided.
+     */
+    public String getLocalInstanceName() {
```

Comment:
rename here and everywhere from "localInstanceName" to "instanceName" and add more details to Java doc saying that this name has no effect on cluster topology and must be set in cases when several nodes are started in a single JVM process or you just need to set some name to all the nodes.